cluster-autoscaler/push_image.sh

```diff
@@ -14,26 +14,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-IMAGE_TO_PUSH=$1
-if [ -z $IMAGE_TO_PUSH ]; then
+if [[ $# -eq 0 ]]; then
   echo No image passed
   exit 1
 fi
 
+IMAGE_TO_PUSH=$1
+shift 1
+
 docker_push_cmd=("docker")
 if [[ "${IMAGE_TO_PUSH}" == "gcr.io/"* ]] || [[ "${IMAGE_TO_PUSH}" == "staging-k8s.gcr.io/"* ]] ; then
     docker_push_cmd=("gcloud" "docker" "--")
 fi
 
-echo "About to push image $IMAGE_TO_PUSH"
+echo "About to push image/manifest ${IMAGE_TO_PUSH}"
 read -r -p "Are you sure? [y/N] " response
 if [[ "$response" =~ ^([yY])+$ ]]; then
-  "${docker_push_cmd[@]}" pull $IMAGE_TO_PUSH
-  if [ $? -eq 0 ]; then
-    echo $IMAGE_TO_PUSH already exists
+  if "${docker_push_cmd[@]}" pull "${IMAGE_TO_PUSH}"; then
+    echo "${IMAGE_TO_PUSH} already exists"
     exit 1
   fi
-  "${docker_push_cmd[@]}" push $IMAGE_TO_PUSH
+  if [[ $# -eq 0 ]]; then
+    "${docker_push_cmd[@]}" push "${IMAGE_TO_PUSH}"
+  else
+    DOCKER_CLI_EXPERIMENTAL=enabled "${docker_push_cmd[@]}" manifest create "${IMAGE_TO_PUSH}" "$@"
+    DOCKER_CLI_EXPERIMENTAL=enabled "${docker_push_cmd[@]}" manifest push --purge "${IMAGE_TO_PUSH}"
+  fi
```

Comment:
Unless things have changed recently, this would require that the arch specific images already be pushed, in which case I think it might make sense to handle the manifest creation/pushing in make rather than here.

Reply:
The push process is a bit strange, as it asks for confirmation that you want to push the image. Not sure if still needed considering that after promoting images from k8s.io they cannot be changed anymore.
I think you are right and the manifest doesn't need these checks as it just uses the already pushed images.